tradeexecutor/analysis/trade_analyser.py

```diff
@@ -332,14 +329,27 @@ class TradeSummary:
     realised_profit: USDollarAmount
     open_value: USDollarAmount
     uninvested_cash: USDollarAmount
-    
+
     initial_cash: USDollarAmount
     extra_return: USDollarAmount
     duration: datetime.timedelta = field(metadata=config(
         encoder=json_encode_timedelta,
         decoder=json_decode_timedelta,
     ))
 
+    average_winning_trade_profit_pc: float
+    average_losing_trade_loss_pc: float
+    biggest_winning_trade_pc: float
+    biggest_losing_trade_pc: float
+    average_duration_of_winning_trades: datetime.timedelta = field(metadata=config(
+        encoder=json_encode_timedelta,
+        decoder=json_decode_timedelta,
+    ))
```

Comment:
Curious, what info does the "duration" give us here? I know it's in the specs, but wouldn't it make more sense to track the maximum length of successive winning trades?

Asking because II can, for example, only make one trade a day, and with two successful trades the duration would be 24 hours. On the other hand, if I trade every hour and make 8 successive win trades, the duration of the streak would "only" be 8 hours, which is much less than 24 hours.

In other words, this value is heavily dominated by the frequency of the trades, meaning it's perhaps not all that informative? @miohtama can you give your thoughts here?

Reply:
As per the spec, it is the average duration of all winning trades combined. But it might be for consecutive successive trades.